### ISSUE TITLE
False positive superTypeTypeParametersChanged

### ISSUE THREAD

**Issue by tmortagne** [revapi/revapi#254]
My understanding is that "superTypeTypeParametersChanged" is currently a "just in case" report which does not currently look at details much and is already knows to potentially lead to false positives, but reporting this false positive use case anyway in case you would like to do something about it at some point.

An example of the failing use case can be seen on:
* the super type: https://github.com/xwiki/xwiki-platform/commit/e14c345aa0487520909b0bb98716e5a3581cac0f#diff-e62bb29be8f50b0ddc53cde0a75278d8e11da66c3afc12445151c3b5ce091c4dR28
* the failing class: https://github.com/xwiki/xwiki-platform/commit/e14c345aa0487520909b0bb98716e5a3581cac0f#diff-70920d9de9b71889a4d01b35034a1857e5454f33299166462eed296fff0a0835R22

To summarize: a new generic parameter was added to the abstract class `NumberProperty` and this generic type is then used as a private field. Revapi report a "superTypeTypeParametersChanged" for all the types which extends `NumberProperty`.

This could definitely lead to semantic breakage (for the same reason `elementNowParameterized` is documented as a potential semantic breakage), but I don't see how this specific change can lead to binary or source breakage.

**Comment by metlos** [comment]
This is true. On the other hand, you can use the power of the java matcher to ignore this kind of change en-masse.

In `revapi.differences` or `revapi.ignore`, add the following ignore declaration:
```
    <item>
      <code>java.class.superTypeTypeParametersChanged</code>
      <ignore>true</ignore>
      <new>
        <matcher>java</matcher>
        <match><!CDATA[
           type ^* extends com.xpn.xwiki.objects.NumberProperty<*>{}
        ]]></match>
      </new>
    </item>
```

(This was written from top of my head. Please experiment with it and read through https://github.com/revapi/classif/blob/master/README.adoc).

**Comment by metlos** [comment]
Btw. of course we can make the detection of super-type type parameter more clever and not report changes that cannot cause the breakage, as the change you've made seems to be (the only thing your change would cause would be a warning about usage of raw types if some user code had a `List<NumberProperty>` for example).

But that would be quite complex to implement and computationally intensive at runtime, IMHO, so I am not sure about the ROI of such smart detection.
